Detectors/TPC/simulation/include/TPCSimulation/SpaceCharge.h

```diff
@@ -61,98 +63,149 @@ class SpaceCharge
   /// Default constructor using a grid size of (129 z bins, 180 phi bins, 129 r bins)
   SpaceCharge();
   /// Constructor with grid size specified by user
-  /// \param nZSlices number of grid points in z, must be (2**N)+1
-  /// \param nPhiBins number of grid points in phi
   /// \param nRBins number of grid points in r, must be (2**N)+1
-  SpaceCharge(int nZSlices, int nPhiBins, int nRBins);
-  /// Constructor with grid size and interpolation order specified by user
-  /// \param nZSlices number of grid points in z, must be (2**N)+1
   /// \param nPhiBins number of grid points in phi
+  /// \param nZSlices number of grid points in z, must be (2**N)+1
+  SpaceCharge(int nRBins, int nPhiBins, int nZSlices);
+  /// Constructor with grid size and interpolation order specified by user
   /// \param nRBins number of grid points in r, must be (2**N)+1
+  /// \param nPhiBins number of grid points in phi
+  /// \param nZSlices number of grid points in z, must be (2**N)+1
   /// \param interpolationOrder order used for interpolation of lookup tables
-  SpaceCharge(int nZSlices, int nPhiBins, int nRBins, int interpolationOrder);
+  SpaceCharge(int nRBins, int nPhiBins, int nZSlices, int interpolationOrder);
 
   // Destructor
   ~SpaceCharge() = default;
```

Comment:
@ehellbar OK for fixing legacy stuff in other PR, but this one was not addressed: you are using default destructor for a class which allocates lot of memory, this will lead to a memory leak. Could you please add a proper destructor?

Reply:
Hello @shahor02 ,
in this class (SpaceCharge.h), I put unique_ptrs where there were raw pointers before, so they all should be destroyed by the default constructor? 
Now I also checked in the legacy classes and modified one of their destructors accordingly.

Reply:
Hi @ehellbar 
thanks, yes, with unique_ptrs it should work, will merge once tests are passed.